x-pack/plugins/cases/common/api/cases/user_actions/operations/find.ts

```diff
@@ -30,8 +30,8 @@ export const UserActionFindRequestRt = rt.exact(
   rt.partial({
     types: rt.array(FindTypeFieldRt),
     sortOrder: rt.union([rt.literal('desc'), rt.literal('asc')]),
-    page: NumberFromString,
-    perPage: NumberFromString,
+    page: rt.union([rt.number, NumberFromString]),
+    perPage: rt.union([rt.number, NumberFromString]),
```

Comment:
I think it would be nice if we create a validation schema for the pagination. For example:

```
export const PaginationSchema = new rt.Type<{ page?: number; perPage?: number }, boolean, unknown>(
  'Test',
  rt.partial({ page: rt.number, perPage: rt.number }).is,
  (u, c) =>
    either.chain(rt.partial({ page: rt.number, perPage: rt.number }).validate(u, c), (params) => {
      if (params.page == null && params.perPage) {
        return rt.success(params);
      }

      const pageAsNumber = params.page ?? 0;
      const perPageAsNumber = params.perPage ?? 0;

      if (Math.max(pageAsNumber, pageAsNumber * perPageAsNumber) > MAX_DOCS_PER_PAGE) {
        return rt.failure(
          u,
          c,
          `The number of documents is too high. Paginating through more than ${MAX_DOCS_PER_PAGE} documents is not possible.`
        );
      }

      return rt.success(params);
    }),
  Boolean
);
```

Reply:
I like it, will do 👍